cwltool/builder.py

```diff
@@ -211,7 +211,13 @@ def bind_input(self,
 
             bp = list(aslist(lead_pos))
             if "position" in binding:
-                bp.extend(aslist(binding["position"]))
+                position = binding["position"]
+                if isinstance(position, str):  # no need to test the CWL Version
+                                               # the schema for v1.0 only allow ints
+                    binding['position'] = self.do_eval(position, context=datum)
```

Comment:
You can simplify this whole thing to just `position = self.do_eval(binding["position"], context=datum)` because if the expression isn't a string it will just return it as-is.

Reply:
but there should be a typecheck on the return value

Reply:
I'd rather keep the `isinstance`